src/compiler/optimizer.js

```diff
@@ -17,13 +18,27 @@ let isPlatformReservedTag
  */
 export function optimize (root: ?ASTElement, options: CompilerOptions) {
   if (!root) return
+  isStaticKey = genStaticKeys(options.modules)
```

Comment:
We are generating a new map every time `optimize` is called. Wondering if there's a way to cache this since most of the time it's going to be the same map.